local-tests/setup/tinny-operations.ts

```diff
@@ -0,0 +1,209 @@
+  );
+
+  // Initialize arrays to keep track of skipped, failed, and passed tests
+  let skippedTests: string[] = [];
+  let failedTests: string[] = [];
+  let passedTests: string[] = [];
+
+  // Iterate over each test and run it in series
+  for (const [testName, testFunction] of testsToRun) {
+    const maxAttempts = devEnv.processEnvs.MAX_ATTEMPTS;
+    let attempts = 0;
+    let testPassed = false;
+
+    while (attempts < maxAttempts && !testPassed) {
+      const startTime = performance.now();
+
+      try {
+        console.log(`Attempt ${attempts + 1} for ${testName}...`);
+
+        // @ts-ignore
+        await testFunction(devEnv);
+        testPassed = true;
+
+        const endTime = performance.now();
+        const timeTaken = (endTime - startTime).toFixed(2);
+        console.log(`${testName} - Passed (${timeTaken} ms)`);
+        passedTests.push(`${testName} (Passed in ${timeTaken} ms)`);
+      } catch (error) {
+        if (error.message === 'LIT_IGNORE_TEST') {
+          skippedTests.push(`${testName} (Skipped)`);
+          break;
+        }
+        attempts++;
+        if (attempts >= maxAttempts) {
+          const endTime = performance.now();
+          const timeTaken = (endTime - startTime).toFixed(2);
+          console.error(
+            `${testName} - Failed after ${maxAttempts} attempts (${timeTaken} ms)`
+          );
+          console.error(`Error: ${error}`);
+          failedTests.push(
+            `${testName} (Failed in ${timeTaken} ms) - Error: ${error}`
+          );
+        }
+      }
+
+      await new Promise((resolve) =>
+        setTimeout(resolve, devEnv.processEnvs.RUN_IN_BAND_INTERVAL)
+      );
+    }
+  }
+
+  passedTests.forEach((test) => console.log(`- ${test}`));
+  failedTests.forEach((test) => console.log(`- ${test}`));
+  skippedTests.forEach((test) => console.log(`- ${test}`));
+
+  console.log();
+  // Print results
+  console.log(
+    `Test Report: ${passedTests.length} test(s) passed, ${failedTests.length} failed, ${skippedTests.length} skipped.`
+  );
+
+  if (failedTests.length > 0) {
+    process.exit(1); // Exit with error code if any test failed
+  } else {
+    process.exit(0); // Exit successfully if all tests passed
+  }
+};
+
+/**
+ * Runs tests in parallel and reports the results.
+ * @param {Object} options - The options for running the tests.
+ * @param {any} options.tests - The tests to run.
+ * @param {TinnyEnvironment} options.devEnv - The development environment.
+ * @returns {Promise<void>} - A promise that resolves when all tests have been run.
+ */
+export const runTestsParallel = async ({
+  tests,
+  devEnv,
+}: {
+  tests: any;
+  devEnv: TinnyEnvironment;
+}): Promise<void> => {
+  const filters = getFiltersFlag();
+  const testsToRun = Object.entries(tests).filter(
+    ([testName]) => filters.length === 0 || filters.includes(testName)
+  );
+
+  const testPromises = testsToRun.map(
+    async ([testName, testFunction], testIndex) => {
```

Comment:
Logging from all of these tests is going to be interleaved if they run in parallel, making it impossible to tell what's really going on especially in the case of failures :(.  Thinking about the effort needed to resolve that issue, though, I think rather than fixing it in this code, we can let `nx` and `Jest` resolve this problem :+1: